src/staking/observable/index.ts

```diff
@@ -0,0 +1,383 @@
+   * @throws {StakingError} - If the delegation is invalid or the transaction cannot be built
+   */
+  public createWithdrawTimelockUnbondedTransaction = (
+    stakingParams: ObservableStakingParams,
+    delegation: ObservableDelegation,
+    feeRate: number,
+  ): PsbtTransactionResult => {
+    validateParams(stakingParams);
+    validateDelegationInputs(
+      delegation, stakingParams, this.stakerInfo,
+    );
+    const { 
+      stakingTx,
+      stakingOutputIndex,
+      timelock,
+      finalityProviderPkNoCoordHex,
+      stakerPkNoCoordHex,
+    } = delegation;
+
+    // Build scripts
+    const scripts = buildScripts(
+      stakingParams,
+      finalityProviderPkNoCoordHex,
+      timelock,
+      stakerPkNoCoordHex,
+    );
+
+    // Create the timelock unbonded transaction
+    try {
+      return withdrawTimelockUnbondedTransaction(
+        scripts,
+        stakingTx,
+        this.stakerInfo.address,
+        this.network,
+        feeRate,
+        stakingOutputIndex,
+      );  
+    } catch (error) {
+      throw StakingError.fromUnknown(
+        error, StakingErrorCode.BUILD_TRANSACTION_FAILURE,
+        "Cannot build unsigned timelock unbonded transaction",
+      );
+    }
+  }
+}
+
+export const validateDelegationInputs = (
+  delegation: ObservableDelegation,
+  stakingParams: ObservableStakingParams,
+  stakerInfo: StakerInfo,
+) => {
+  const { stakingTx, startHeight, timelock, stakingOutputIndex } = delegation;
+  if (startHeight < stakingParams.activationHeight) {
+    throw new StakingError(
+      StakingErrorCode.INVALID_INPUT,
+      "Staking transaction start height cannot be less than activation height",
+    );
+  }
+
+  if (
+    timelock < stakingParams.minStakingTimeBlocks ||
+    timelock > stakingParams.maxStakingTimeBlocks
+  ) {
+    throw new StakingError(
+      StakingErrorCode.INVALID_INPUT,
+      "Staking transaction timelock is out of range",
+    );
+  }
+
+  if (delegation.stakerPkNoCoordHex !== stakerInfo.publicKeyNoCoordHex) {
+    throw new StakingError(
+      StakingErrorCode.INVALID_INPUT,
+      "Staker public key does not match between connected staker and delegation staker",
+    );
+  }
+
+  if (!isValidNoCoordPublicKey(delegation.finalityProviderPkNoCoordHex)) {
+    throw new StakingError(
+      StakingErrorCode.INVALID_INPUT,
+      "Finality provider public key should not have a coordinate",
+    );
+  }
+
+  if (!stakingTx.outs[stakingOutputIndex]) {
+    throw new StakingError(
+      StakingErrorCode.INVALID_INPUT,
+      "Staking transaction output index is out of range",
+    );
+  }
+
+  if (stakingTx.getId() !== delegation.stakingTxHashHex) {
+    throw new StakingError(
+      StakingErrorCode.INVALID_INPUT,
+      "Staking transaction hash does not match between the btc transaction and the provided staking hash",
+    );
+  }
+}
+
+export const buildScripts = (
+  params: ObservableStakingParams,
+  finalityProviderPkNoCoordHex: string,
+  stakingTerm: number,
```

Comment:
We are using the terms stakingTerm and timelock interchangeably which might lead to confusion. Have added this in a previous review, couldn't find the response, adding this here to make sure it wasn't missed.